classes/key_manager.php

```diff
@@ -0,0 +1,144 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Access key manager.
+ *
+ * @package    tool_pdfpages
+ * @author     Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright  2021 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_pdfpages;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Access key manager.
+ *
+ * @package    tool_pdfpages
+ * @author     Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright  2021 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class key_manager {
```

Comment:
just wondering why did you decide to make all methods of this class static?

Reply:
So you don't have to instantiate it, there is no data stored in the class, it's just an API

Reply:
not all classes need to store data to be instantiated. I won't be picky here. So it's up to you to rewrite. Personally I'd prefer not to use static functions where possible.